### PR TITLE
Owners can submit waivers

### LINKED ISSUE #49: Owners can submit waiver requests for their team
When an owner want to submit a new waiver request, they go to their team page and submit it.

### PR DESCRIPTION
**Pull request by axelclark**
Add waiver form, with new and create actions
    * Select from owned players to drop
    * Select from available players to add
    * Refactored javascript filter to use classes
    * Refactored views to move common functions in to view_helpers
    * Used a nested route so admin can submit waivers for other teams
    ** Gets team_id from the route rather than the current user
    * Link only visible to owners and admins
    * When a new waiver is created, the default is "pending"
    * Added pending to test helper
    * Authorizes only owners and admins to submit waivers
    * Closes #49

### CODE REVIEW

File: web/controllers/waiver_controller.ex [hunk 1]
```diff
@@ -15,4 +21,62 @@ defmodule Ex338.WaiverController do
...
+        conn
+        |> put_flash(:info, "Waiver successfully submitted.")
+        |> redirect(to: fantasy_team_path(conn, :show, team_id))
+      {:error, changeset} ->
+        fantasy_team = FantasyTeam
+                        |> Repo.get(team_id)
+        fantasy_league = FantasyLeague
+                          |> Repo.get(fantasy_team.fantasy_league_id)
+        owned_players = FantasyTeam.owned_players(team_id)
+                        |> Repo.all
```
Comment (credoci) [inline web/controllers/waiver_controller.ex]: Pipe chain should start with a raw value.

File: web/controllers/waiver_controller.ex [hunk 2]
```diff
@@ -15,4 +21,62 @@ defmodule Ex338.WaiverController do
...
+    fantasy_league = FantasyLeague |> Repo.get(fantasy_team.fantasy_league_id)
+
+    changeset =
+      fantasy_team
+      |> build_assoc(:waivers)
+      |> Waiver.changeset
+
+
+    owned_players = FantasyTeam.owned_players(team_id)
+                    |> Repo.all
```
Comment (credoci) [inline web/controllers/waiver_controller.ex]: Pipe chain should start with a raw value.

File: web/controllers/waiver_controller.ex [hunk 3]
```diff
@@ -15,4 +21,62 @@ defmodule Ex338.WaiverController do
...
+
+    avail_players = FantasyPlayer.available_players(fantasy_league.id)
+                    |> Repo.all
```
Comment (credoci) [inline web/controllers/waiver_controller.ex]: Pipe chain should start with a raw value.

File: web/controllers/waiver_controller.ex [hunk 4]
```diff
@@ -15,4 +21,62 @@ defmodule Ex338.WaiverController do
...
+      {:error, changeset} ->
+        fantasy_team = FantasyTeam
+                        |> Repo.get(team_id)
+        fantasy_league = FantasyLeague
+                          |> Repo.get(fantasy_team.fantasy_league_id)
+        owned_players = FantasyTeam.owned_players(team_id)
+                        |> Repo.all
+
+        avail_players = FantasyPlayer.available_players(fantasy_league.id)
+                        |> Repo.all
```
Comment (credoci) [inline web/controllers/waiver_controller.ex]: Pipe chain should start with a raw value.

File: web/models/waiver.ex [hunk 5]
```diff
@@ -19,7 +19,17 @@ defmodule Ex338.Waiver do
     struct
     |> cast(params, [:status, :fantasy_team_id, :add_fantasy_player_id,
                      :drop_fantasy_player_id])
-    |> validate_required([:status])
+    |> validate_required([:fantasy_team_id])
+  end
+
+  def new_changeset(struct, params \\ %{}) do
```
Comment (credoci) [inline web/models/waiver.ex]: Parameter `struct` has same name as the `Kernel.struct` function.

**Comment by axelclark** [comment]
Closed before git push.

**Pull request merged**